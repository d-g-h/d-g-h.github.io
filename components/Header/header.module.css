.name {
  font-size: 1.5rem;
  margin: 0 0 0.25rem 0;
}

.links {
  display: flex;
  flex-direction: column;
  margin-bottom: 0.75rem;

  @media (min-width: 640px) {
    justify-content: space-between;
    flex-direction: row;
  }
}
