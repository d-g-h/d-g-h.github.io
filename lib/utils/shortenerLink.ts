export default function shortenerLink({
  url,
  link,
}: {
  url: string;
  link: string;
}): string {
  try {
    const shortenedURL = new URL(url);
    return `${link}${shortenedURL.pathname}`;
  } catch (error) {
    console.log(error);
    return "";
  }
}
