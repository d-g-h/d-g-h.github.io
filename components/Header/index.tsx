type HeaderProps = {
  phone?: boolean;
  text?: string;
};

import shortenerLink from "@/lib/utils/shortenerLink";
import styles from "@/components/Header/header.module.css";

export default function Header({ phone, text }: Readonly<HeaderProps>) {
  return (
    <>
      <header>
        <div className={styles.container}>
          {text ? (
            <div
              dangerouslySetInnerHTML={{ __html: text }}
              aria-label="Generated SVG"
            />
          ) : (
            ""
          )}
          <h1 className={styles.name}>
            <a
              href={`${process.env.NEXT_PUBLIC_GITHUB}`}
            >{`${process.env.NEXT_PUBLIC_NAME}`}</a>
          </h1>
        </div>
        <div className={`${styles.sub} ${styles.links}`}>
          <div>
            <a href={`mailto:${process.env.NEXT_PUBLIC_EMAIL}`}>
              {process.env.NEXT_PUBLIC_EMAIL}
            </a>
          </div>
          <div>
            <a href={`${process.env.NEXT_PUBLIC_LINKEDIN}`}>
              {shortenerLink({
                url: `${process.env.NEXT_PUBLIC_LINKEDIN}`,
                link: `linkedin.com`,
              })}
            </a>
          </div>
          <div>
            <a href={`${process.env.NEXT_PUBLIC_GITHUB}`}>
              {shortenerLink({
                url: `${process.env.NEXT_PUBLIC_GITHUB}`,
                link: `github.com`,
              })}
            </a>
          </div>
          {phone ? (
            <div>
              <a href={`tel:+1${process.env.NEXT_PUBLIC_PHONE}`}>
                {process.env.NEXT_PUBLIC_PHONE}
              </a>
            </div>
          ) : (
            ""
          )}
        </div>
      </header>
    </>
  );
}
