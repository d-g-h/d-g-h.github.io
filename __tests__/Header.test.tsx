/**
 * @jest-environment jsdom
 */
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Header from "@/components/Header";
import shortenerLink from "@/lib/utils/shortenerLink";

const env = {
  NEXT_PUBLIC_URL: "https://qr.qr/t",
  NEXT_PUBLIC_NAME: "Name Name",
  NEXT_PUBLIC_GITHUB: "https://github.com/profile",
  NEXT_PUBLIC_EMAIL: "email@email.email",
  NEXT_PUBLIC_LINKEDIN: "https://linkedin.com/in/profile",
  NEXT_PUBLIC_PHONE: "7188675309",
};

beforeAll(() => {
  Object.assign(process.env, env);
});

afterAll(() => {
  jest.restoreAllMocks();
});

describe("Header", () => {
  it("renders Header component correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText(env.NEXT_PUBLIC_NAME)).toBeInTheDocument();
    expect(screen.getByText(env.NEXT_PUBLIC_NAME).closest("a")).toHaveAttribute(
      "href",
      env.NEXT_PUBLIC_GITHUB,
    );

    expect(screen.getByText(env.NEXT_PUBLIC_EMAIL)).toBeInTheDocument();
    expect(screen.getByText(env.NEXT_PUBLIC_EMAIL).closest("a")).toHaveAttribute(
      "href",
      `mailto:${env.NEXT_PUBLIC_EMAIL}`,
    );

    expect(
      screen.getByText(shortenerLink({ url: env.NEXT_PUBLIC_LINKEDIN, link: "linkedin.com" })),
    ).toBeInTheDocument();
    expect(
      screen
        .getByText(shortenerLink({ url: env.NEXT_PUBLIC_LINKEDIN, link: "linkedin.com" }))
        .closest("a"),
    ).toHaveAttribute("href", env.NEXT_PUBLIC_LINKEDIN);

    expect(
      screen.getByText(shortenerLink({ url: env.NEXT_PUBLIC_GITHUB, link: "github.com" })),
    ).toBeInTheDocument();
    expect(
      screen
        .getByText(shortenerLink({ url: env.NEXT_PUBLIC_GITHUB, link: "github.com" }))
        .closest("a"),
    ).toHaveAttribute("href", env.NEXT_PUBLIC_GITHUB);

    expect(screen.queryByText(env.NEXT_PUBLIC_PHONE)).not.toBeInTheDocument();
  });

  it("renders phone link when phone prop is true", () => {
    const { container } = render(<Header phone />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText(env.NEXT_PUBLIC_PHONE)).toBeInTheDocument();
    expect(screen.getByText(env.NEXT_PUBLIC_PHONE).closest("a")).toHaveAttribute(
      "href",
      `tel:+1${env.NEXT_PUBLIC_PHONE}`,
    );
  });

  it("renders qr code when url is defined", async () => {
    const { container } = render(<Header text={env.NEXT_PUBLIC_URL} />);
    expect(container).toMatchSnapshot();
  });
});
