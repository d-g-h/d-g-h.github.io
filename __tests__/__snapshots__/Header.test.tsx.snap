// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header component correctly 1`] = `
<div>
  <header>
    <div
      class="container"
    >
      <h1
        class="name"
      >
        <a
          href="https://github.com/profile"
        >
          Name Name
        </a>
      </h1>
    </div>
    <div
      class="sub links"
    >
      <div>
        <a
          href="mailto:email@email.email"
        >
          email@email.email
        </a>
      </div>
      <div>
        <a
          href="https://linkedin.com/in/profile"
        >
          linkedin.com/in/profile
        </a>
      </div>
      <div>
        <a
          href="https://github.com/profile"
        >
          github.com/profile
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header renders phone link when phone prop is true 1`] = `
<div>
  <header>
    <div
      class="container"
    >
      <h1
        class="name"
      >
        <a
          href="https://github.com/profile"
        >
          Name Name
        </a>
      </h1>
    </div>
    <div
      class="sub links"
    >
      <div>
        <a
          href="mailto:email@email.email"
        >
          email@email.email
        </a>
      </div>
      <div>
        <a
          href="https://linkedin.com/in/profile"
        >
          linkedin.com/in/profile
        </a>
      </div>
      <div>
        <a
          href="https://github.com/profile"
        >
          github.com/profile
        </a>
      </div>
      <div>
        <a
          href="tel:+17188675309"
        >
          7188675309
        </a>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header renders qr code when url is defined 1`] = `
<div>
  <header>
    <div
      class="container"
    >
      <img
        alt="url QR code"
        class="img"
        data-nimg="1"
        decoding="async"
        height="100"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fqr.qr%2Ft&w=256&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fqr.qr%2Ft&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fqr.qr%2Ft&w=256&q=75 2x"
        style="color: transparent;"
        width="100"
      />
      <h1
        class="name"
      >
        <a
          href="https://github.com/profile"
        >
          Name Name
        </a>
      </h1>
    </div>
    <div
      class="sub links"
    >
      <div>
        <a
          href="mailto:email@email.email"
        >
          email@email.email
        </a>
      </div>
      <div>
        <a
          href="https://linkedin.com/in/profile"
        >
          linkedin.com/in/profile
        </a>
      </div>
      <div>
        <a
          href="https://github.com/profile"
        >
          github.com/profile
        </a>
      </div>
    </div>
  </header>
</div>
`;
